lib/parser/source/tree_rewriter.rb

```diff
@@ -361,6 +366,28 @@ def insert_after_multi(range, text)
 
       private
 
+      def action_summary
+        replacements = as_replacements
+        case replacements.size
+        when 0 then return 'empty'
+        when 1..3 then #ok
+        else
+          replacements = replacements.first(3)
+          suffix = '…'
+        end
+        parts = replacements.map do |(range, str)|
+          if str.empty?
```

Comment:
Not sure if I understand what happens in this if/elsif logic? Could you explain it please?

Reply:
The `replacements` method returns a range and the string that replaces it. The if/elsif/ logic breaks these into deletions, additions, and replacements.

Reply:
got it, thanks. is there a better way to identify these replacements?

Reply:
actually, I think it's enough to have these comments, 👍  if there's no other way to do it (without rewriting)

Reply:
Sorry, not sure what you mean

Reply:
the way how you jump into if/else branches is not very clean and explicit. Technically it's based on undocumented behavior, and it can break once insertion has a non-empty `range` (for absolutely any reason). I was thinking about using some better check to identify replacements.

Reply:
Internally, these are all treated as replacements. It's simply that humans will see a replacement of a range corresponding to "hello" with an empty string as a deletion. But technically there is no distinction between `delete(range)` and `replace(range, '')`. Same idea of straight insertion vs  replacement.
In short: this can not break, because a straight insertion is always computed as a string replacing an empty range, by definition.